### PR TITLE
when reading time series without a time-window get all data

### PR DESCRIPTION
**Pull request by ktarbet**

### CODE REVIEW

File: heclib/hecdss/hecdss.c [hunk 1]
```diff
@@ -280,6 +280,17 @@ HECDSS_API int hec_dss_tsRetrieve(dss_file* dss, const char *pathname,
...
+
+    // if no dates/times are given retrieve all data.
+    if (  
+         (startDate == NULL || startDate == "\0")
+      && (startTime == NULL || startTime == "\0")
+      && (endDate == NULL   || endDate == "\0")
+      && (endTime == NULL   || endTime == "\0")
+         ) 
+    {
+      tss->boolRetrieveAllTimes = 1;
```
Comment (perrymanmd) [inline heclib/hecdss/hecdss.c:292]: Shouldn't this be only if the D-part of pathname is empty? Normal behavior is to just get the single block if the D-parts is specified and there is not time window.

Comment (ktarbet) [reply heclib/hecdss/hecdss.c:292]: Thanks for that question, I forgot about that.

This API is aspiring to work for a future back-end where blocks may not exist.  
I propose we ignore the D part to make the API cleaner; Then we have one way to specify a time-window.

**Review by perrymanmd — APPROVED** [review]

File: heclib/hecdss/hecdss.c (hunk 1 above)
Comment (MikeNeilson) [reply heclib/hecdss/hecdss.c:292]: I think that is a breaking change, perhaps a good one, but given the D part is shown in tools like DSSVue should there be at least a warning message somewhere to help prevent quite confusion?

**Review by MikeNeilson — COMMENTED** [review]
I like it, one less thing to consider, but may be breaking in a bad? way.

**Pull request merged**